test/unit/search/search_scheme_algorithm_test.cpp

```diff
@@ -0,0 +1,329 @@
+        for (std::vector<uint8_t> & resElem : error_distributions[search_id])
+            order_search_vector(resElem, search_scheme[search_id]);
+        max_error = std::max(max_error, search_scheme[search_id].u.back());
+    }
+
+    for (uint64_t text_length = 10; text_length < 10000; text_length *= 10)
+    {
+        uint64_t const query_length_min = std::max<uint64_t>(3, search_scheme.front().blocks() * max_error);
+        uint64_t const query_length_max = std::min<uint64_t>(16, text_length);
+
+        random_text(text, text_length);
+        bi_fm_index index(text);
+
+        for (uint64_t i = 0; i < iterations; ++i)
+        {
+            for (uint64_t query_length = query_length_min; query_length < query_length_max; ++query_length)
+            {
+                auto const block_info = search_scheme_block_info(search_scheme, query_length);
+                for (uint8_t search_id = 0; search_id < search_scheme.size(); ++search_id)
+                {
+                    auto const & [blocks_length, start_pos] = block_info[search_id];
+
+                    std::vector<uint64_t> ordered_blocks_length;
+                    get_ordered_search(search_scheme[search_id], blocks_length,
+                                       ordered_search_scheme[search_id], ordered_blocks_length);
+
+                    for (auto && error_distribution : error_distributions[search_id])
+                    {
+                        test_search_hamming(index.begin(), text, search_scheme[search_id], query_length,
+                                            error_distribution, seed, blocks_length, ordered_blocks_length, start_pos);
+                    }
+                }
+            }
+        }
+    }
+}
+
+template <typename search_scheme_t>
+inline void test_search_scheme_edit(search_scheme_t const & search_scheme, time_t const seed, uint64_t const iterations)
+{
+    using text_t = dna4_vector;
+
+    text_t text, query;
+
+    // retrieve maximum number of errors from search_scheme
+    uint8_t max_error = 0;
+    for (auto const & search : search_scheme)
+        max_error = std::max(max_error, search.u.back());
+
+    for (uint64_t text_length = 10; text_length < 10000; text_length *= 10)
+    {
+        uint64_t const query_length_min = std::max<uint64_t>(3, search_scheme.front().blocks() * max_error);
+        uint64_t const query_length_max = std::min<uint64_t>(16, text_length);
+
+        random_text(text, text_length);
+        bi_fm_index index(text);
+
+        uint8_t const substitution = std::rand() % (max_error + 1);
+        uint8_t const insertion    = std::rand() % (max_error + 1);
+        uint8_t const deletion     = std::rand() % (max_error + 1);
+        detail::search_param error_left{max_error, substitution, insertion, deletion};
+
+        for (uint64_t i = 0; i < iterations; ++i)
+        {
+            for (uint64_t query_length = query_length_min; query_length < query_length_max; ++query_length)
```

Comment:
You are using `uint64_t` as counter variable types everywhere. Is this related to the structure of the index or could it be `size_t`, as well?